File: app/models/sso_user_binding.go
package models

import "time"

type SSOUserBinding struct {
	ID             uint64    `gorm:"column:id;primaryKey" json:"id"`
	UserID         uint64    `gorm:"column:user_id" json:"user_id"`
	ProviderID     uint64    `gorm:"column:provider_id" json:"provider_id"`
	SSOUserID      string    `gorm:"column:sso_user_id" json:"sso_user_id"`
	SSOEmail       string    `gorm:"column:sso_email" json:"sso_email"`
	SSOUsername    string    `gorm:"column:sso_username" json:"sso_username"`
	SSOAvatar      string    `gorm:"column:sso_avatar" json:"sso_avatar"`
	AccessToken    string    `gorm:"column:access_token" json:"-"`
	RefreshToken   string    `gorm:"column:refresh_token" json:"-"`
	TokenExpiresAt time.Time `gorm:"column:token_expires_at" json:"token_expires_at"`
	FirstLoginAt   time.Time `gorm:"column:first_login_at" json:"first_login_at"`
	LastLoginAt    time.Time `gorm:"column:last_login_at" json:"last_login_at"`
	LoginCount     int       `gorm:"column:login_count" json:"login_count"`
	Timestamps
}

func (SSOUserBinding) TableName() string {
	return "sso_user_binding"
}

File: MineAdmin-web/src/modules/base/api/ssoUserBinding.ts
import type { PageList, ResponseStruct } from '#/global'

export interface SSOUserBindingVo {
  id: number
  user_id: number
  username?: string
  nickname?: string
  provider_id: number
  provider_name?: string
  provider_display_name?: string
  provider_type?: string
  provider_scene?: string
  sso_user_id: string
  sso_email?: string
  sso_username?: string
  sso_avatar?: string
  login_count: number
  first_login_at?: string
  last_login_at?: string
  token_expires_at?: string
  created_at?: string
  updated_at?: string
}

export interface SSOUserBindingSearchVo {
  page?: number
  page_size?: number
  user_id?: number
  username?: string
  provider_id?: number
  provider_name?: string
  sso_user_id?: string
  sso_email?: string
  sso_username?: string
}

export function page(data: SSOUserBindingSearchVo): Promise<ResponseStruct<PageList<SSOUserBindingVo>>> {
  return useHttp().get('/admin/sso-user-binding/list', { params: data })
}

export function detail(id: number): Promise<ResponseStruct<SSOUserBindingVo>> {
  return useHttp().get(`/admin/sso-user-binding/${id}`)
}

export function userBindings(userId: number): Promise<ResponseStruct<SSOUserBindingVo[]>> {
  return useHttp().get(`/admin/sso-user-binding/user/${userId}`)
}

export function unbind(id: number): Promise<ResponseStruct<null>> {
  return useHttp().delete(`/admin/sso-user-binding/${id}`)
}
